packages/core/src/search/search.ts

```diff
@@ -49,45 +49,46 @@ export interface IncludeTarget {
  * These operators represent "modifiers" and "prefixes" in FHIR search.
  * See: https://www.hl7.org/fhir/search.html
  */
-export enum Operator {
-  EQUALS = 'eq',
-  NOT_EQUALS = 'ne',
+export const Operator = {
+  EQUALS: 'eq' as const,
+  NOT_EQUALS: 'ne' as const,
 
   // Numbers
-  GREATER_THAN = 'gt',
-  LESS_THAN = 'lt',
-  GREATER_THAN_OR_EQUALS = 'ge',
-  LESS_THAN_OR_EQUALS = 'le',
+  GREATER_THAN: 'gt' as const,
+  LESS_THAN: 'lt' as const,
+  GREATER_THAN_OR_EQUALS: 'ge' as const,
+  LESS_THAN_OR_EQUALS: 'le' as const,
 
   // Dates
-  STARTS_AFTER = 'sa',
-  ENDS_BEFORE = 'eb',
-  APPROXIMATELY = 'ap',
+  STARTS_AFTER: 'sa' as const,
+  ENDS_BEFORE: 'eb' as const,
+  APPROXIMATELY: 'ap' as const,
 
   // String
-  CONTAINS = 'contains',
-  STARTS_WITH = 'sw',
-  EXACT = 'exact',
+  CONTAINS: 'contains' as const,
+  STARTS_WITH: 'sw' as const,
+  EXACT: 'exact' as const,
 
   // Token
-  TEXT = 'text',
-  NOT = 'not',
-  ABOVE = 'above',
-  BELOW = 'below',
-  IN = 'in',
-  NOT_IN = 'not-in',
-  OF_TYPE = 'of-type',
+  TEXT: 'text' as const,
+  NOT: 'not' as const,
+  ABOVE: 'above' as const,
+  BELOW: 'below' as const,
+  IN: 'in' as const,
+  NOT_IN: 'not-in' as const,
+  OF_TYPE: 'of-type' as const,
 
   // All
-  MISSING = 'missing',
-  PRESENT = 'present',
+  MISSING: 'missing' as const,
+  PRESENT: 'present' as const,
 
   // Reference
-  IDENTIFIER = 'identifier',
+  IDENTIFIER: 'identifier' as const,
 
   // _include and _revinclude
-  ITERATE = 'iterate',
-}
+  ITERATE: 'iterate' as const,
+};
```

Comment:
We should probably make the whole object as const, I think that will fix the cast issue